pandas/io/json.py

```diff
@@ -884,3 +939,174 @@ def _recursive_extract(data, path, seen_meta, level=0):
         result[k] = np.array(v).repeat(lengths)
 
     return result
+
+
+# ---------------------------------------------------------------------
+# JSON-Table Schema routines
+# http://specs.frictionlessdata.io/json-table-schema/
+
+
+def as_json_table_type(x):
+    """
+    Convert a NumPy / pandas type to its corresponding json_table.
+
+    Parameters
+    ----------
+    x : array or dtype
+
+    Returns
+    -------
+    t : str
+        the Table Schema data types
+
+    Notes
+    -----
+    This table shows the relationship between NumPy / pandas dtypes,
+    and Table Schema dtypes.
+
+    ==============  =================
+    Pandas type     Table Schema type
+    ==============  =================
+    int64           integer
+    float64         number
+    bool            boolean
+    datetime64[ns]  date
+    timedelta64[ns] duration
+    object          str
```

Comment:
Hmm, the name `is_string_dtype` seems a bit misleading :-)

```
In [9]: pd.types.common.is_string_dtype(pd.Series([datetime.date(2012,1,1)]))
Out[9]: True
```

Looking at the code, it indeed just returns True for all object dtypes.

I just noted the final `else: return 'any'`, and supposed that was for none string dtypes.

I suppose you have to infer the type to be able to distuinguish real strings:

```
In [11]: pd.lib.infer_dtype(pd.Series(['a', 'b']))
Out[11]: 'string'

In [12]: pd.lib.infer_dtype(pd.Series([datetime.date(2012, 1, 2)]))
Out[12]: 'date'

In [13]: pd.lib.infer_dtype(pd.Series([object()]))
Out[13]: 'mixed'
```

but not sure if that is worth it here. Question is maybe also what to do with those non-string object dtypes? They would get there string repr anyway in json?